internal/cache/cache.go

```diff
@@ -75,119 +77,117 @@ func NewLoadingCache(options ...Option) (*LoadingCache, error) {
 func (c *LoadingCache) Set(key string, value interface{}) {
 	c.Lock()
 	defer c.Unlock()
-
 	now := time.Now()
-	if _, ok := c.data[key]; !ok {
-		c.data[key] = &cacheItem{}
-	}
-	c.data[key].data = value
-	c.data[key].expiresAt = now.Add(c.ttl)
-	if c.isLRU {
-		c.data[key].lastReadAt = now
+
+	// Check for existing item
+	if ent, ok := c.data[key]; ok {
```

Comment:
What the reason to move to front an item for existing key on set? If I understand the reason correctly (provide proper sorting order in case of TTL) why do we do it for LRU as well?

Reply:
It's logic of LRU we currently use. We want to update both creation and lastRead time to mark an entry as newest.

If we insert new entry which updates the oldest entry by lastRead, and won't update lastRead (position in list in this case), it will be deleted on next DeleteOldest call despite being the newest inserted entry.

Reply:
yeah, i think it makes sense